modules/s3-bucket/main.tf

```diff
@@ -1,53 +1,83 @@
-  kms_master_key_arn                     = var.kms_master_key_arn
-  lifecycle_rules                        = var.lifecycle_rules
-  logging                                = var.logging
-  object_lock_configuration              = var.object_lock_configuration
-  policy                                 = local.bucket_policy
-  privileged_principal_actions           = var.privileged_principal_actions
-  privileged_principal_arns              = var.privileged_principal_arns
-  restrict_public_buckets                = var.restrict_public_buckets
-  s3_replica_bucket_arn                  = var.s3_replica_bucket_arn
-  s3_replication_enabled                 = var.s3_replication_enabled
-  s3_replication_rules                   = var.s3_replication_rules
-  s3_replication_source_roles            = var.s3_replication_source_roles
-  sse_algorithm                          = var.sse_algorithm
-  transfer_acceleration_enabled          = var.transfer_acceleration_enabled
-  versioning_enabled                     = var.versioning_enabled
-  website_inputs                         = var.website_inputs
+  version = "2.0.0"
+
+  bucket_name = var.bucket_name
+
+  # Object access and permissions
+  acl                          = var.acl
+  grants                       = var.grants
+  allow_encrypted_uploads_only = var.allow_encrypted_uploads_only
+  allow_ssl_requests_only      = var.allow_ssl_requests_only
+  block_public_acls            = var.block_public_acls
+  block_public_policy          = var.block_public_policy
+  ignore_public_acls           = var.ignore_public_acls
+  restrict_public_buckets      = var.restrict_public_buckets
+  logging                      = var.logging
+  source_policy_documents      = [local.bucket_policy]
+  privileged_principal_actions = var.privileged_principal_actions
+  privileged_principal_arns    = var.privileged_principal_arns
+
+  # Static website configuration
+  cors_rule_inputs = var.cors_rule_inputs
+  website_inputs   = var.website_inputs
+
+  # Bucket feature flags
+  transfer_acceleration_enabled = var.transfer_acceleration_enabled
+  versioning_enabled            = var.versioning_enabled
+  force_destroy                 = var.force_destroy
+  object_lock_configuration     = var.object_lock_configuration
+
+  # Object lifecycle rules
+  lifecycle_configuration_rules = var.lifecycle_configuration_rules
+
+  # Object encryption
+  sse_algorithm      = var.sse_algorithm
+  kms_master_key_arn = var.kms_master_key_arn
+
+  # Object replication
+  s3_replication_enabled      = var.s3_replication_enabled
+  s3_replica_bucket_arn       = var.s3_replica_bucket_arn
+  s3_replication_rules        = var.s3_replication_rules
+  s3_replication_source_roles = var.s3_replication_source_roles
 
   # IAM user with permissions to access the s3 bucket
   user_enabled           = var.user_enabled
   allowed_bucket_actions = var.allowed_bucket_actions
 
-  context = module.this.context
+  context = module.introspection.context
```

Comment:
Let's not use `introspection` in `terraform-aws-components`, at least not yet.

Reply:
removed